pkg/yurtadm/cmd/join/join.go

```diff
@@ -320,6 +327,21 @@ func newJoinData(args []string, opt *joinOptions) (*joinData, error) {
 		return nil, err
 	}
 	data.kubernetesVersion = k8sVersion
+
+	// check whether specified nodePool exists
+	if len(opt.nodePoolName) != 0 {
+		yurtClient, err := kubeconfigutil.ToYurtClientSet(cfg)
```

Comment:
Why do we need to write a new `ToYurtClientSet` method to get the client? We should have already gotten the client in [here](https://github.com/openyurtio/openyurt/blob/master/pkg/yurtadm/cmd/join/join.go#LL310C2-L310C2)

Reply:
This place need a nodepool client to get information, the normal client cannot do it.